pkg/controller/infrastructure/actuator.go

```diff
@@ -643,7 +651,15 @@ func (a *actuator) cleanupServiceLoadBalancers(ctx context.Context, infra *exten
 }
 
 // Delete implements infrastructure.Actuator.
-func (a *actuator) Delete(ctx context.Context, log logr.Logger, infra *extensionsv1alpha1.Infrastructure, _ *extensioncontroller.Cluster) error {
+func (a *actuator) Delete(ctx context.Context, log logr.Logger, infra *extensionsv1alpha1.Infrastructure, cluster *extensioncontroller.Cluster) error {
+	if a.shouldUseFlow(infra, cluster) {
```

Comment:
how about there is no infra status but shoot/seed was labeled?

Reply:
Code changed, only check flow state when do Delete